File: reactor/java-reactor/src/main/java/com/example/reactor/javareactor/router/UserRouter.java
package com.example.reactor.javareactor.router;

import com.example.reactor.javareactor.handler.UserHandler;
import lombok.RequiredArgsConstructor;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.reactive.function.server.RouterFunction;
import org.springframework.web.reactive.function.server.RouterFunctions;
import org.springframework.web.reactive.function.server.ServerResponse;

import static org.springframework.web.reactive.function.server.RequestPredicates.GET;
import static org.springframework.web.reactive.function.server.RequestPredicates.path;

@RequiredArgsConstructor
@Configuration
public class UserRouter {
    @Bean
    public RouterFunction<ServerResponse> userRoutes(UserHandler handler) {
        return RouterFunctions.nest(
            path("/user"),
            RouterFunctions.route(GET("/{id}"), handler::getUserById)
        );
    }
}

File: reactor/kotlin-reactor/src/main/kotlin/com/example/reactor/kotlinreactor/router/UserRouter.kt
package com.example.reactor.kotlinreactor.router

import com.example.reactor.kotlinreactor.handler.UserHandler
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.web.reactive.function.server.RouterFunction
import org.springframework.web.reactive.function.server.ServerResponse
import org.springframework.web.reactive.function.server.router

@Configuration
class UserRouter {
    @Bean
    fun userRoutes(handler: UserHandler): RouterFunction<ServerResponse> =
        router {
            "/user".nest {
                GET("/{id}", handler::getUserById)
            }
        }
}
